lib/clients/http_client.js

```diff
@@ -628,6 +629,15 @@ HttpClient.prototype._options = function (method, options) {
         opts.socketPath = this.socketPath;
     }
 
+    // Add the request path to the base path of the client
+    // Don't combine the request url and client path if they both
+    // have a slash
+    if (opts.path && this.url.path &&
+        opts.path !== '/' && this.url.path !== '/') {
+        // clean up the url path, stripping the last /
+        opts.path = utils.sanitizePath([this.url.path, opts.path].join('/'));
```

Comment:
This wouldn't affect #692 as this is a http client making requests to any other http server / endpoint, not the server managing routes.

`url.resolve()` won't work if the request url has a / at the beginning, it will replace whatever path is already at the beginning of the path. 

```
url.resolve('/foo/bar', '/baz') === '/baz';
```